reporters/allure-testinfo-adapter.js

```
@@ -0,0 +1,27 @@
+// reporters/allure-testinfo-adapter.js
+const { allure } = require('allure-playwright');
+
+/** @type {import('@playwright/test/reporter').Reporter} */
+class AllureTestInfoAdapter {
+  onTestBegin(test, testInfo) {
+    // TEMP: prove the reporter runs
+    // console.log('[adapter] inject for:', test.title);
+
+    testInfo.allure = {
+      parentSuite: (...a) => allure.parentSuite(...a),
+      suite:       (...a) => allure.suite(...a),
+      subSuite:    (...a) => allure.subSuite(...a),
+      epic:        (...a) => allure.epic(...a),
+      feature:     (...a) => allure.feature(...a),
+      story:       (...a) => allure.story(...a),
+      severity:    (...a) => allure.severity(...a),
+      label:       (...a) => allure.label(...a),
+      link:        (...a) => allure.link(...a),
+      id:          (...a) => allure.id(...a),
+      owner:       (...a) => allure.owner(...a),
+      tag:         (...a) => allure.tag(...a),
+      parameter:   (...a) => allure.parameter(...a),
+    };
```

This block unconditionally overwrites `testInfo.allure` with functions that call the module-level `allure` (a singleton). That is exactly the kind of fallback/override the corrections prohibit. Quote of required correction: **"HIGH: Remove any fallback from testInfo.allure to a module-level singleton for attaching Allure metadata (e.g., code like const a = testInfo?.allure ?? singleton). All metadata operations must use the Playwright adapter bound to the current test."**

Overriding `testInfo.allure` here will: 1) detach metadata from the Playwright-bound per-test adapter, and 2) mask reporter/misconfiguration issues. Instead, do not override `testInfo.allure`; if you need to assert availability, check for it and throw when missing (see next comment).